### ISSUE TITLE
USB emulation is no longer supported

### ISSUE THREAD

**Issue by knuxyl** [LineageOS/android_kernel_motorola_msm8998#1]
Why has USB emulation been deleted from newest LineageOS builds? Having support for DriveDroid is a must have for me and putting together an entire build environment and praying I did everything right is a huge hassle. Could someone please implement this into the kernel? Thanks!

I'm using a Moto X4 RETUS (payton) SDM630 phone (source for it says it uses this kernel)

Here is what the developer of DriveDroid suggested to make it work

https://github.com/CyanogenMod/android_kernel_lge_hammerhead/commit/46bd47757a9f34c1d95dd2620414893212fb0526#diff-cb60cc142bbe2e7d1b5b9bee23329377

**Comment by daedroza** [comment]
At least people working on SDM660 have not touched this.
If possible, could you flash both old/new version and provide us logs?

**Comment by knuxyl** [comment]
Well I'm referring to older lineageos on a different phone. I assumed stuff like this would be generic across the board for all kernels. I cant flash this one because it's my daily driver. I can share the logcat of when I try to mount the image though, but itll have to be later

**Comment by daedroza** [comment]
Nope, it is not generic at all. It requires kernel patches and won't be supported here unless there is a good reason to. You're better off with older lineage/different phone.

Regarding your commit, USB/CD switch is completely unnecessary, maybe you need to tweak your USB options in Settings.

Additionally, you can check on 4pda/xda if anyone has this working on Android 9 (4.4.xxx kernel).
